elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -0,0 +1,415 @@
+                file_lead_path = os.path.join(base_path, 'tests/ans.py')
+                with open(file_lead_path, 'r') as file:
+                    solution_file = file.read()
+                return f'<pl-code language="python">{solution_file}</pl-code>'
+            except FileNotFoundError:
+                return 'The reference solution is not provided for this question.'
+
+        grading_mode = pl.get_string_attrib(element, 'grading-method', 'ordered')
+        grading_mode = 'in any order' if grading_mode == 'unordered' else 'in the specified order'
+
+        check_indentation = pl.get_boolean_attrib(element, 'indentation', INDENTION_DEFAULT)
+        check_indentation = ', with correct indentation' if check_indentation is True else None
+
+        if answer_name in data['correct_answers']:
+            html_params = {
+                'true_answer': True,
+                'question_solution': pretty_print(data['correct_answers'][answer_name]['correct_answers']),
+                'grading_mode': grading_mode,
+                'check_indentation': check_indentation
+            }
+            with open('pl-order-blocks.mustache', 'r', encoding='utf-8') as f:
+                html = chevron.render(f, html_params)
+            return html
+        else:
+            return ''
+
+
+def pretty_print(array):
+    if array is None:
+        return None
+    pretty_print_answer = []
+    for text in array:
+        temp = {'text': text}
+        pretty_print_answer.append(dict(temp))
+    return pretty_print_answer
+
+
+def parse(element_html, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    answer_name = pl.get_string_attrib(element, 'answers-name')
+
+    temp = answer_name
+    temp += '-input'
+    # the answer_name textfields that raw-submitted-answer reads from
+    # have '-input' appended to their name attribute
+
+    student_answer_temp = ''
+    if temp in data['raw_submitted_answers']:
+        student_answer_temp = data['raw_submitted_answers'][temp]
+
+    if student_answer_temp is None:
+        data['format_errors'][answer_name] = 'NULL was submitted as an answer.'
+        return
+    elif student_answer_temp == '':
+        data['format_errors'][answer_name] = 'No answer was submitted.'
+        return
+
+    student_answer = []
+    student_answer_indent = []
+    grading_mode = pl.get_string_attrib(element, 'grading-method', GRADING_METHOD_DEFAULT)
+
+    student_answer_ranking = ['Question grading_mode is not "ranking"']
+
+    student_answer_temp = json.loads(student_answer_temp)
+
+    student_answer = student_answer_temp['answers']
+    student_answer_indent = student_answer_temp['answer_indent']
+
+    if grading_mode.lower() == 'ranking':
+        student_answer_ranking = []
+        pl_drag_drop_element = lxml.html.fragment_fromstring(element_html)
+        for answer in student_answer:
+            e = pl_drag_drop_element.xpath(f'.//pl-answer[text()="{answer}"]')
```

Comment:
I've stated in previous comments (which now disappeared many edits later) that xpath here is overkill, we could do some much simpler and more efficient string comparison. 

However, this is a hidden detail we can always improve later so I don't think we should worry about it right now.

Reply:
I do remember you bringing this up, and I think I might have forgotten to get back to you about the use of xpath, but xpath is used because we need to parse the actual html element so pl.get_element_string_attribute would work. There is definitely a cleaner way to do this, but like you said we'll probably improve this later. Ive also just added a note in #3476